x-pack/platform/plugins/shared/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/ml_vcu_usage_cost_tour.tsx

```diff
@@ -0,0 +1,102 @@
+import { EuiButtonEmpty, EuiLink, EuiText, EuiTourStep } from '@elastic/eui';
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useStateWithLocalStorage } from '../../../../../hooks/use_state_with_localstorage';
+import { useAppContext } from '../../../../../app_context';
+
+export interface MlVcuUsageCostTourProps {
+  anchorPosition?: EuiTourStepProps['anchorPosition'];
+  children: React.ReactElement;
+}
+
+export const MlVcuUsageCostTour = ({
+  anchorPosition = 'downCenter',
+  children,
+}: MlVcuUsageCostTourProps) => {
+  const [isTourVisible, setTourVisible] = useStateWithLocalStorage<boolean>(
+    'MlVcuUsageCostTourSkipKey',
+    false
+  );
+  const { docLinks } = useAppContext();
+
+  return (
+    <EuiTourStep
+      title={
+        <FormattedMessage
+          id="xpack.idxMgmt.mappingsEditor.parameters.mlCostTour.title"
+          defaultMessage="Machine learning usage"
+        />
+      }
+      subtitle={
+        <FormattedMessage
+          id="xpack.idxMgmt.mappingsEditor.parameters.mlCostTour.subtitle"
+          defaultMessage="VCU consumption"
+        />
+      }
+      maxWidth="500px"
+      content={
+        <EuiText>
+          <p>
+            <FormattedMessage
+              id="xpack.idxMgmt.mappingsEditor.parameters.mlCostTour.contents"
+              values={{
+                additionalCostsLink: (
+                  <EuiLink
+                    data-test-subj="mlVcuUsageAdditionalCostLink"
+                    target="_blank"
+                    href={docLinks.links.observability.elasticServerlessSearchManagedLlmUsageCost}
+                  >
+                    <FormattedMessage
+                      id="xpack.idxMgmt.mappingsEditor.parameters.mlCostTour.additionalCosts"
+                      defaultMessage="additional costs"
+                    />
+                  </EuiLink>
+                ),
+                learnMoreLink: (
+                  <EuiLink
+                    data-test-subj="mlVcuUsageLearnMoreLink"
+                    target="_blank"
+                    href={docLinks.links.cloud.elasticsearchBillingManagingCosts}
+                  >
+                    <FormattedMessage
+                      id="xpack.idxMgmt.mappingsEditor.parameters.mlCostTour.learnMore"
+                      defaultMessage="Learn more"
+                    />
+                  </EuiLink>
+                ),
+              }}
+              defaultMessage="The VCUs used to perform inference, NLP tasks, and other ML activities will incur {additionalCostsLink}. You can continue to use other models as normal. {learnMoreLink}"
+            />
+          </p>
+        </EuiText>
+      }
+      isStepOpen={!isTourVisible}
+      anchorPosition={anchorPosition}
+      step={1}
+      stepsTotal={1}
+      onFinish={() => setTourVisible(true)}
+      data-test-subj="mlVcuUsageCostTour"
+      footerAction={
+        <EuiButtonEmpty
```

Comment:
I'm seeing a `EuiButtonEmpty must include an accessible label. Use one of: aria-label, aria-labelledby` warning in VS Code. Should we add an accessible label here given that there's a focus on accessibility in our products?

Reply:
added in https://github.com/elastic/kibana/pull/235264/commits/4427f46964c2c7a3ca10cf85db183dfe429accc8